### PR TITLE
Reaper provision expansion

### PR DESCRIPTION
**Pull request by LtChae**
+ This is in support of a data ingest card
+ Extends the timestamp provider with a version 2 that takes format, offset, and timezone
+ Allows hosted file ingestions to use providers. (This was not required for the card but it will save us from ourselves later)

### CODE REVIEW

File: apps/reaper/lib/reaper/file_ingest/processor.ex [hunk 1]
```diff
@@ -17,7 +18,8 @@ defmodule Reaper.FileIngest.Processor do
   @doc """
   Process a hosted dataset
   """
-  def process(%SmartCity.Dataset{} = dataset) do
+  def process(%SmartCity.Dataset{} = unprovisioned_dataset) do
+    dataset = Provisioner.provision(unprovisioned_dataset)
```
Comment (dunson062786) [inline apps/reaper/lib/reaper/file_ingest/processor.ex:22]: What does Provisioner.provision do?

File: apps/providers/test/unit/providers/date_test.exs [hunk 2]
```diff
@@ -32,7 +33,9 @@ defmodule Providers.DateTest do
 
       result = Providers.Date.provide("1", %{format: format, offset_in_days: offset_in_days})
 
-      assert Timex.now() |> Timex.add(Timex.Duration.from_days(offset_in_days)) |> Timex.format!(format) == result
+      assert Timex.now()
```
Comment (bbrewer-papa-x) [inline apps/providers/test/unit/providers/date_test.exs:36]: what if this runs right at midnight? :D

Comment (LtChae) [reply apps/providers/test/unit/providers/date_test.exs:36]: Dunno. I suspect its not worth trying to solve that particular edge case
